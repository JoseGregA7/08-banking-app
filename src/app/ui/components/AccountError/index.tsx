import { NavigateFunction } from "react-router-dom";
import './style.scss';

const AccountError = ({ error, navigate }: { error: string | null, navigate: NavigateFunction }) => {
    return (
        <div
            style={{ padding: '20px', fontFamily: 'Arial, sans-serif' }}
            role="alert"
            aria-live="assertive"
            data-testid="account-error-container"
        >
            <h2
                title="Account Section"
                aria-label="Account Overview"
            >
                Mi Cuenta
            </h2>
            {error && (
                <div
                    role="region"
                    aria-labelledby="error-message"
                    data-testid="account-error-message"
                >
                    <div
                        id="error-message"
                        aria-live="assertive"
                        aria-atomic="true"
                    >
                        Parece que a√∫n no tienes una cuenta.
                    </div>
                    <button
                        className="button"
                        onClick={() => navigate('/create-account')}
                        title="Crear cuenta"
                        aria-label="Crear una cuenta"
                        data-testid="create-account-button"
                    >
                        Crear una cuenta
                    </button>
                </div>
            )}
        </div>
    );
};

export default AccountError;
