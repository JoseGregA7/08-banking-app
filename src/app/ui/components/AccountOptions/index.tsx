import './style.scss';
import SubOptions from '../AccountSubOptions';
const AccountOptions = ({ selectedOption, handleOptionClick, handleClose, handleWithdrawTypeChange, handleWithdrawalAmountChange, handleWithdrawalSubmit, state, handleDepositSubmit, handleDepositAmountChange, handleDepositTypeChange, handleBuySubmit, handleBuyAmountChange, handleBuyTypeChange }: { selectedOption: string | null, handleOptionClick: (option: string) => void, handleClose: () => void, handleWithdrawTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleWithdrawalAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleWithdrawalSubmit: (e: React.FormEvent) => void, state: any, handleDepositTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleDepositAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleDepositSubmit: (e: React.FormEvent) => void, handleBuyTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleBuyAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleBuySubmit: (e: React.FormEvent) => void }) => {
  return (
    <div className="Option__wrapper">
      <div className="Option__title">Opciones de la Cuenta</div>
      <div className="Option__container">
        <button name="option_button_dep贸sitos" role="option_button_dep贸sitos" aria-label="option_button_dep贸sitos" className="Option__button" onClick={() => handleOptionClick('depositos')}>Dep贸sitos</button>
        <button name="option_button_retiros" role="option_button_retiros" aria-label="option_button_retiros" className="Option__button" onClick={() => handleOptionClick('retiros')}>Retiros</button>
        <button name="option_button_compras" role="option_button_compras" aria-label="option_button_compras" className="Option__button" onClick={() => handleOptionClick('compras')}>Compras</button>
      </div>
      {
        selectedOption &&
        <div className='Option__close'>
          <div>
            <button name="cerrar" role="button" aria-label="Cerrar" onClick={handleClose}>x</button>
          </div>
        </div>
      }
      {selectedOption && <SubOptions option={selectedOption} handleWithdrawTypeChange={handleWithdrawTypeChange} handleWithdrawalAmountChange={handleWithdrawalAmountChange} handleWithdrawalSubmit={handleWithdrawalSubmit} state={state} handleDepositTypeChange={handleDepositTypeChange} handleDepositAmountChange={handleDepositAmountChange} handleDepositSubmit={handleDepositSubmit} handleBuyTypeChange={handleBuyTypeChange} handleBuyAmountChange={handleBuyAmountChange} handleBuySubmit={handleBuySubmit} />}
    </div>
  );
};

export default AccountOptions;
