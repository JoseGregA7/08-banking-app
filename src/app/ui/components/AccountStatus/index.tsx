import { IAccountInfo } from '../../../core/interfaces/state';
import './style.scss';

interface AccountStatusProps {
    accountInfo: IAccountInfo | null;
}

const labelMapping = {
    number: 'Número de Cuenta',
    amount: 'Saldo Actual',
    createdAt: 'Fecha de Creación',
};

const AccountStatus = ({ accountInfo }: AccountStatusProps ) => {
    const filteredData = accountInfo ? Object.entries(accountInfo)
        .filter(([key, value]) => value !== null)
        .map(([key, value]) => {
            const label = labelMapping[key as keyof typeof labelMapping] || key;
            return [label, value];
        }) : [];

    return (
        <div className='AccountStatus__wrapper'>
            <div className='AccountStatus__main'>
                <div className='AccountStatus__title'>Estado de la Cuenta</div>
                <div className='AccountStatus__content'>
                    {filteredData.length === 0 ? (
                        <div>No hay datos disponibles.</div>
                    ) : (
                        <ul className='AccountStatus__list'>
                            {filteredData.map(([label, value]) => (
                                <li className='AccountStatus__item' key={label}>
                                    <strong>{label}:</strong> {value}
                                </li>
                            ))}
                        </ul>
                    )}
                </div>
            </div>
        </div>
    );
};

export default AccountStatus;
