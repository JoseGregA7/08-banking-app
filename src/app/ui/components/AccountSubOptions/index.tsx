import BuyOptions from "../BuyOptions";
import DepositOptions from "../DepositOptions";
import WithdrawalOptions from "../WithdrawalOptions";

const SubOptions = ({ option, handleWithdrawTypeChange, handleWithdrawalAmountChange, handleWithdrawalSubmit, state, handleDepositSubmit, handleDepositAmountChange, handleDepositTypeChange, handleBuySubmit, handleBuyAmountChange, handleBuyTypeChange }: { option: string, handleWithdrawTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleWithdrawalAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleWithdrawalSubmit: (e: React.FormEvent) => void, state: any, handleDepositTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleDepositAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleDepositSubmit: (e: React.FormEvent) => void, handleBuyTypeChange: (e: React.ChangeEvent<HTMLSelectElement>) => void, handleBuyAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void, handleBuySubmit: (e: React.FormEvent) => void }) => {
  switch (option) {
    case 'depositos':
      return <DepositOptions handleDepositTypeChange={handleDepositTypeChange} handleDepositAmountChange={handleDepositAmountChange} handleDepositSubmit={handleDepositSubmit} state={state} />;
    case 'retiros':
      return <WithdrawalOptions handleWithdrawTypeChange={handleWithdrawTypeChange} handleWithdrawalAmountChange={handleWithdrawalAmountChange} handleWithdrawalSubmit={handleWithdrawalSubmit} state={state} />;
    case 'compras':
      return <BuyOptions handleBuyTypeChange={handleBuyTypeChange} handleBuyAmountChange={handleBuyAmountChange} handleBuySubmit={handleBuySubmit} state={state} />;
    default:
      return null;
  }
};

export default SubOptions;