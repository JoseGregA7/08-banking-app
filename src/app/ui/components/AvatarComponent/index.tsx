import './style.scss';

const AvatarLogout = ({handleEndSession}:{handleEndSession:()=>void}) => {
    return (
        <div className='avatar-logout'>
            <img src="avatar.webp" alt="Avatar" onClick={handleEndSession} className="avatar" />
        </div>
    );
};

export default AvatarLogout;
