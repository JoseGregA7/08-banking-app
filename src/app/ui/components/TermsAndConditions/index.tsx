import './style.scss';

export const TermsAndConditions = () => {
    return (
        <div>
            <a href="#" className='home__terms-conditions'>
                TÃ©rminos y Condiciones
            </a>
        </div>
    )
};