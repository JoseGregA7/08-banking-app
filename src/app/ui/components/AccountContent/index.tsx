import { NavigateFunction } from 'react-router-dom';
import './style.scss';
import AccountStatus from '../AccountStatus';


const AccountContent = ({ navigate, account }: { navigate: NavigateFunction, account: { amount: number, id: number, name: string } }) => {
    return (
        <div className='AccountContent__container'>
            <AccountStatus accountInfo={account} />
            <div className='AccountContent__divider'>|
                <button className='button' onClick={() => navigate('/')}>Transacciones</button>
                <span className='AccountContent__divider'>|</span>
                <button className='button' onClick={() => navigate('/operations')}>Operaciones</button>
            </div>
        </div>)
}
export default AccountContent;