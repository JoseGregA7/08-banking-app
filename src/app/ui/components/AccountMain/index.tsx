import AccountContent from '../AccountContent';
import AccountError from '../AccountError';
import './style.scss';

const AccountMain = ({ navigate, accountInfo, loading, error }: { navigate: any, accountInfo: any, loading: boolean, error: string | null }) => {
    console.log('accountInfo', accountInfo);
    return (
        <div className='AccountMain__wrapper'>
            <div className='AccountMain__title'>Mi Cuenta</div>
            {error && (
                <AccountError error={error} navigate={navigate} />
            )}
            {loading && <p>Cargando...</p>}
            {accountInfo && !loading && !error && (
                <AccountContent navigate={navigate} account={accountInfo} />
            )}
        </div>
    )
}

export default AccountMain;