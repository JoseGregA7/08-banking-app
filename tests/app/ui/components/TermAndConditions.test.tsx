import React from 'react';
import { TermsAndConditions } from '../../../../src/app/ui/components/TermsAndConditions';
import { describe, expect, it, test } from 'vitest';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';


describe('TermAndConditions', () => {
  test('should render the component', () => {
    render(<TermsAndConditions />);
    expect(document.body).toHaveTextContent('Términos y Condiciones');
  });
  test('match snapshot', () => {
    const { asFragment } = render(<TermsAndConditions />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should have a link to the terms and conditions', () => {
    const { getByText } = render(<TermsAndConditions />);
    expect(getByText('Términos y Condiciones')).toBeInTheDocument();
  });
  test('Shouñd have a className of home__terms-conditions', () => {
    const { getByText } = render(<TermsAndConditions />);
    expect(getByText('Términos y Condiciones')).toHaveClass('home__terms-conditions');
  });
});