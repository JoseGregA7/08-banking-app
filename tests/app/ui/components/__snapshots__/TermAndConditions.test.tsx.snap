// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TermAndConditions > match snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="home__terms-conditions"
      href="#"
    >
      TÃ©rminos y Condiciones
    </a>
  </div>
</DocumentFragment>
`;
